### PR TITLE
MUSL deltas (mostly harmless)

### PR DESCRIPTION
**Pull request by KenDickey**
I have built and tested aarch64 builds on
  Raspbery Pi 3B / Arch Linux / MUSL+Busybox
  Chromebook Two / Linux / libc

The mvm "-O1" seems to work in both cases -- no need to go to "-O0"

Please ignore the spurious sqSCCSVersion.h -- I don't know how to elide this.

The getwd() -> getcwd() in sqUnixSecurityPlugin.c should likewise be harmless [no getwd()  in MUSL]

Not a time concern.  I don't know of anyone asking for this.

### CODE REVIEW

**Comment by krono** [comment]
>  Please ignore the spurious sqSCCSVersion.h -- I don't know how to elide this.

This is explained [in the README](https://github.com/OpenSmalltalk/opensmalltalk-vm#important-notice-for-developers):
> ## Important notice for Developers:
>
> We rely on source file substitutions in the src tree, specifically, any sq*SCCSVersion.h files anywhere in the tree are processed to replace $Rev$, $Date$, and $URL$ with the current revision (defined as the timestamp %Y%m%d%H%M of the latest commit), the human readable date of that commit, and the url of the origin repository these sources were cloned from.
>
> The first time you clone this repository, you must therefore run this command:
>
>     ./scripts/updateSCCSVersions
>
> This will install filters, post-commit, and post-merge hooks to update the sq*SCCSVersion.h files anytime you commit or merge.

**Comment by KenDickey** [comment]
Tobias,

Thanks for the help.

More details.

Got messed up somewhere.  Symptom is
````
Alpine:aarch64:~ >>> cd OpenSmalltalk/oscogvm/
Alpine:aarch64:~/OpenSmalltalk/oscogvm >>> scripts/updateSCCSVersions 
error: patch failed: platforms/Cross/plugins/sqPluginsSCCSVersion.h:9
error: platforms/Cross/plugins/sqPluginsSCCSVersion.h: patch does not apply
error: patch failed: platforms/Cross/vm/sqSCCSVersion.h:28
error: platforms/Cross/vm/sqSCCSVersion.h: patch does not apply
Alpine:aarch64:~/OpenSmalltalk/oscogvm >>> 
````

Recloned the KenDickey/opensmalltalk-vm as kens-vm, 
ran the scripts/updateSCCSVersion
and all shows well.

But how do I get git to tell GitHub to clean this?

Thanks much again,
-KenD

**Comment by krono** [comment]
(sorry, short on Time atm.)

**Comment by KenDickey** [comment]
On 2019-12-15 23:27, Tobias Pape wrote:

> (sorry, short on Time atm.)

Ah!  Sorry here too.  You can ignore this latest.

I had not realize the push would impact the MUSL pull request.

FYI, what I am working toward is getting vm-display-fbdev up using 
libevdev.  Basically moving from the ol' PS2 drivers to USB drivers for 
mouse and keyboard.  Part of this is mutating the stand-alone evtest.c 
code to enable/show the connection to the SqueakVM mouse and keyboard 
api.

Of course, I have to understand what is there first.  Bottom up working 
code lets me travel there a bit at a time so to speak.  ;^)

Oh, yea.  Ignore the guy behind the curtain!
-KenD

File: platforms/Cross/plugins/sqPluginsSCCSVersion.h [hunk 1]
```diff
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 201911192350 $";
```
Comment (nicolas-cellier-aka-nice) [inline platforms/Cross/plugins/sqPluginsSCCSVersion.h]: As Tobias noticed, the changes to this file should be reverted

File: platforms/Cross/vm/sqSCCSVersion.h [hunk 2]
```diff
@@ -28,13 +28,13 @@
 
 #if SUBVERSION
 # define PREFIX "r"
-static char SvnRawRevisionString[] = "$Rev$";
+static char SvnRawRevisionString[] = "$Rev: 201911192350 $";
```
Comment (nicolas-cellier-aka-nice) [inline platforms/Cross/vm/sqSCCSVersion.h:31]: Idem, please revert the changes to this file

File: platforms/unix/plugins/SecurityPlugin/sqUnixSecurity.c [hunk 3]
```diff
@@ -213,7 +213,8 @@ ioInitSecurity(void)
   if (imagePathLen)
     strncpy(secureUserDirectory, imageName, imagePathLen);
   else {
-    getwd(secureUserDirectory);
+/*@@MUSL@@*/
+    getcwd(secureUserDirectory, imagePathLen);
```
Comment (nicolas-cellier-aka-nice) [inline platforms/unix/plugins/SecurityPlugin/sqUnixSecurity.c]: Does it work? We're passing `imagePathLen==0` here.
Shouldn't it be the buffer size, `MAXPATHLEN`?
Or just use more robust `sizeof(secureUserDirectory)`.
Theoretically, the return code might be `NULL` and we should check that too...

**Comment by KenDickey** [comment]
On 2019-12-24 01:51, Nicolas Cellier wrote:

> @NICOLAS-CELLIER-AKA-NICE commented on this pull request.
> -------------------------
> 
> In platforms/Cross/vm/sqSCCSVersion.h:
> 
>> @@ -28,13 +28,13 @@
> 
> #if SUBVERSION
> # define PREFIX "r"
> -static char SvnRawRevisionString[] = "$Rev$";
> +static char SvnRawRevisionString[] = "$Rev: 201911192350 $";
> 
> Idem, please revert the changes to this file

Yes.  I noted in the earlier post that I don't know how to do this 
(trivial git user).

How would I do this?

Thanks,
-KenD

**Comment by KenDickey** [comment]
Greetings all,

The vm-display-fbdev (a.k.a. framebuffer display) has been integrated 
into opensmalltalk-vm.
Thanks Eliot!

This has been tested with both Squeak and Cuis images on amd64 and 
arm64/aarch64.

This requires Linux framebuffer support (/dev/fb0) and libevdev 
('/usr/include/libevdev-1.0/libevdev/libevdev.h').

Hardware tested:
   LePotato (aml-s905x-cc; Armbian Linux; libc)
   Raspberry Pi 3 & 4 (Alpine Linux; MUSL)
   A very old Dell amd64 box (Alpine Linux; MUSL)

Right now the MUSL builds require manual help.

E.g.
   cd build.linux64ARMv8/squeak.cog.spur/build
   ./mvm
   y   ## answer to 'clean?'
--> breaks on vm-display-fbdev build
   cd vm-display-fbdev
   make
   cd ..
   ./mvm
   n  ## answer NO to 'clean?'

The build then proceeds to completion and checks out fine.

Does anyone out there know about how to get 'automake' 
('platforms/unix/config/configure' script) to do the right thing here?

Desired goal is for
   -DMUSL
and
   -DUSEEVDEV
CFLAGS to be set and have the 'mvm' script make vm-display-fbdev without 
manual intervention.

Thanks a bunch for any help!
-KenD

**Comment by krono** [comment]
I think with the landing of #515, most things here are obsolete, and others need different treatment.
Mind if I  close?

**Comment by KenDickey** [comment]
On 2020-09-10 11:15, Tobias Pape wrote:

> I think with the landing of #515, most things here are obsolete, and 
> others need different treatment.
> Mind if I close?

Yep.  Ole news.  Please close.

Thanks,
-KenD
